### PR TITLE
more general result about higher-dimensional quotients

### PR DESCRIPTION
**Pull request by jonsterling**

### CODE REVIEW

**Comment by jonsterling** [comment]
(btw I'm waiting to hear from mike as to whether this is stupid or interesting)

also, the "quotients" really ought to be set-truncated. don't have a nice way of doing that right now.

**Comment by ecavallo** [comment]
i declare it interesting

**Comment by jonsterling** [comment]
:sunglasses:

File: library/quotient.red [hunk 1]
```diff
@@ -11,7 +11,7 @@ data (A : type) (R : A → A → type) ⊢ quotient where
   | i=1 → pt b
   ]
 
--- A quotient by a type-theoretic equivalence relation *with coherences* is strongly effective.
+-- A "quotient" by a "0-coherent groupoid" is effective
```
Comment (cangiuli) [inline library/quotient.red]: :sunglasses:

Comment (jonsterling) [reply library/quotient.red]: taste it

File: library/path-retract.red [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+-- a retract of the path family is an equivalence
+
+-- a path retract always preserves refl, but you might have a more efficient proof
+def path-retract/preserves-refl→equiv (A : type) (R : A → A → type)
+  (ret : (x y : A) → retract (R x y) (path A x y)) (a b : A)
+  (preserves-refl : path _ (ret a a .fst (ret a a .snd .fst refl)) refl)
+  : equiv (R a b) (path A a b)
+  =
+  let s : (x y : A) → R x y → path A x y = λ x y → ret x y .fst in
```
Comment (jonsterling) [inline library/path-retract.red]: slicker to write the following and avoid duplication in these definitions:

```
let (x y : A) : R x y ...
```
```

File: library/path-retract.red [hunk 3]
```diff
@@ -11,10 +11,10 @@ def path-retract/preserves-refl→equiv (A : type) (R : A → A → type)
...
   : equiv (R a b) (path A a b)
   =
-  let s : (x y : A) → R x y → path A x y = λ x y → ret x y .fst in
-  let r : (x y : A) → path A x y → R x y = λ x y → ret x y .snd .fst in
-  let α : (x y : A) (t : R x y) → path (R x y) (r x y (s x y t)) t =
-    λ x y → ret x y .snd .snd
+  let s (x y : A) : R x y → path A x y = ret x y .fst in
+  let r (x y : A) : path A x y → R x y = ret x y .snd .fst in
+  let α (x y : A) : (t : R x y) → path (R x y) (r x y (s x y t)) t =
+    ret x y .snd .snd
```
Comment (jonsterling) [inline library/path-retract.red]: ^ cool ^

**Comment by jonsterling** [comment]
should we merge this?

**Comment by ecavallo** [comment]
yes, let's

**Pull request merged**